core/src/main/java/greencity/service/impl/FactOfTheDayServiceImpl.java

```diff
@@ -27,14 +34,17 @@
  * @author Mykola Lehkyi
  */
 @AllArgsConstructor
-@Service
 @EnableCaching
+@Service
+@Scope(proxyMode = ScopedProxyMode.TARGET_CLASS)
 public class FactOfTheDayServiceImpl implements FactOfTheDayService {
     private final FactOfTheDayRepo factOfTheDayRepo;
     private final ModelMapper modelMapper;
     private final LanguageService languageService;
     private final FactOfTheDayTranslationService factOfTheDayTranslationService;
     private final PlaceCommentServiceImpl placeCommentServiceImpl;
+    @Resource
+    private FactOfTheDayService self;
```

Comment:
What is the problem you are trying to solve here? Why not use `this` instead?

Reply:
https://stackoverflow.com/questions/16899604/spring-cache-cacheable-not-working-while-calling-from-another-method-of-the-s

Reply:
The cache doesn`t work from a method inside the same class.

Reply:
![image](https://user-images.githubusercontent.com/29165636/93628678-99bcab00-f9ef-11ea-9dea-92739aaac227.png)